docs: enhance README.md with comprehensive documentation (#3673)
ava-labs/avalanchego: README.md

A node needs to catch up to the latest network state before it can participate in consensus and serve API calls. This process (called bootstrapping) currently takes several days for a new node connected to Mainnet.

> A node will not [report healthy](https://docs.avax.network/api-reference/health-api) until it is done bootstrapping.

Reviewer: this is a helpful suggestion, If you want to open a new PR to make this change, I would be happy to approve it.

### Running mock codegen

> See [the Contributing document autogenerated mocks section](CONTRIBUTING.md#Autogenerated-mocks).

Reviewer: ```suggestion
See the autogenerated mocks section of the [contributions](CONTRIBUTING.md#Autogenerated-mocks) doc.
```

**We and our community welcome responsible disclosures.**

Please refer to our [Security Policy](SECURITY.md) and [Security Advisories](https://github.com/ava-labs/avalanchego/security/advisories).

> ## Contributing
> 
> We welcome contributions from the community! Here's how you can contribute:
> 
> 1. Fork the repository
> 2. Create your feature branch (`git checkout -b feature/amazing-feature`)
> 3. Commit your changes (`git commit -m 'Add some amazing feature'`)
> 4. Push to the branch (`git push origin feature/amazing-feature`)
> 5. Open a Pull Request
> 
> ### Development Guidelines
> - Follow Go best practices and coding standards
> - Add unit tests for new features
> - Update documentation as needed
> - Run linters before submitting PR
> - Keep commits atomic and well-documented
> 

Reviewer: ```suggestion
```

Most of this information exists in our contributing doc already

> ### Testing
> Before submitting a PR, ensure:

Reviewer: ```suggestion
### Testing

Before submitting a PR, ensure:
```
